sklearn/utils/estimator_checks.py

```diff
@@ -112,7 +112,12 @@ def _yield_checks(estimator):
             # We skip pairwise because the data is not pairwise
             yield check_sample_weights_shape
             yield check_sample_weights_not_overwritten
-            yield check_sample_weight_equivalence
+            yield check_sample_weight_equivalence_on_dense_data
+            # FIXME: filter on tags.input_tags.sparse
+            # (estimator accepts sparse arrays)
+            # once issue #30139 is fixed.
+            if SPARSE_ARRAY_PRESENT:
+                yield check_sample_weight_equivalence_on_sparse_data
```

Comment:
why do we need sparse arrays ? Wouldn't it make more sense to move this if condition into `check_sample_weight_equivalence_on_sparse_data` and pass a csr_array if possible or a csr_matrix otherwise ?